src/dune_pkg/opamUrl0.ml

```diff
@@ -101,8 +103,8 @@ let fetch_revision t ~loc resolve rev_store =
   | Unresolved o ->
     Rev_store.fetch_object rev_store remote o
     >>| (function
-     | None -> not_found t
-     | Some rev -> Ok rev)
+     | Error git_output -> not_found ~git_output t
```

Comment:
Maybe you can also pass this `loc` to `not_found` and then we'll have a decent location in the error as well?